### PR TITLE
Replace transfig with fig2dev

### PR DESCRIPTION
**Pull request by azuwis**
###### Motivation for this change

Reasons:

- transfig never get upstream update since it was added to nixpkgs 2006,
  according to [fig2dev Debian changelog][1], upstream maintainership
  was taken over at 2016, and renamed to fig2dev.

- [Debian][2], [ArchLinux][3], [HomeBrew][4] all use fig2dev to replace
  transfig.

- transfig builds [failed][5] on aarch64-darwin, fig2dev builds fine.

- One less package to maintain.

[1]: https://salsa.debian.org/debian/fig2dev/-/blob/debian/1%253.2.6_beta-1/debian/changelog
[2]: https://packages.debian.org/bullseye/transfig
[3]: https://archlinux.org/packages/community/x86_64/fig2dev/
[4]: https://formulae.brew.sh/formula/fig2dev
[5]: https://hydra.nixos.org/build/163404782/nixlog/2

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by LeSuisse — APPROVED** [review]
Change looks good to me and it builds fine:

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>26 packages built:</summary>
  <ul>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>btrbk</li>
    <li>chemtool</li>
    <li>clevis</li>
    <li>collectd</li>
    <li>collectd-data</li>
    <li>dblatexFull</li>
    <li>disorderfs</li>
    <li>fig2dev</li>
    <li>fped</li>
    <li>kakounePlugins.rep</li>
    <li>libvmi</li>
    <li>luksmeta</li>
    <li>qemu_xen (qemu_xen_4_10)</li>
    <li>qemu_xen-light (qemu_xen_4_10-light)</li>
    <li>qubes-core-vchan-xen</li>
    <li>skribilo</li>
    <li>spring</li>
    <li>springLobby</li>
    <li>tang</li>
    <li>udiskie</li>
    <li>xen</li>
    <li>xen-light</li>
    <li>xen-slim</li>
    <li>xfig</li>
  </ul>
</details>

**Comment by reckenrode** [comment]
Result of `nixpkgs-review pr 156420` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by reckenrode** [comment]
Result of `nixpkgs-review pr 156420` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>btrbk</li>
    <li>collectd</li>
    <li>collectd-data</li>
    <li>qemu_xen</li>
    <li>qemu_xen-light</li>
    <li>qemu_xen_4_10</li>
    <li>qemu_xen_4_10-light</li>
    <li>udiskie</li>
  </ul>
</details>
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>dblatexFull</li>
    <li>fig2dev (transfig)</li>
    <li>rep (kakounePlugins.rep)</li>
    <li>xfig</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -10870,9 +10870,7 @@ with pkgs;
 
   ts = callPackage ../tools/system/ts { };
 
-  transfig = callPackage ../tools/graphics/transfig {
-    libpng = libpng12;
-  };
+  transfig = fig2dev;
```
[the remark this answers is not in the document]
Comment (azuwis) [reply pkgs/top-level/all-packages.nix:10873]: Done

**Comment by azuwis** [comment]
I don't understand the test errors, the only change is https://github.com/NixOS/nixpkgs/pull/156420/commits/9c4bd0a205ddbeccb4d10be9013a4fd05c3232f6, which is quite small.

**Comment by LeSuisse** [comment]
The issue is this:
```
++ nix-env -f /nix/store/9wvzqbggrn9zcnpds8hdmpi7xvirpfi3-source --show-trace --argstr system aarch64-darwin --arg config '{ allowAliases = false; }' --option experimental-features no-url-literals -qa --drv-path --system-filter '*' --system --option build-users-group ''
error: anonymous function at /nix/store/9wvzqbggrn9zcnpds8hdmpi7xvirpfi3-source/pkgs/development/tools/parsing/ragel/default.nix:1:1 called without required argument 'transfig'
```

You need to update `ragel` dependency.

**Comment by azuwis** [comment]
> The issue is this:
> 
> ```
> ++ nix-env -f /nix/store/9wvzqbggrn9zcnpds8hdmpi7xvirpfi3-source --show-trace --argstr system aarch64-darwin --arg config '{ allowAliases = false; }' --option experimental-features no-url-literals -qa --drv-path --system-filter '*' --system --option build-users-group ''
> error: anonymous function at /nix/store/9wvzqbggrn9zcnpds8hdmpi7xvirpfi3-source/pkgs/development/tools/parsing/ragel/default.nix:1:1 called without required argument 'transfig'
> ```
> 
> You need to update `ragel` dependency.

I'm confused, `transfig = fig2dev` in `aliases.nix`  works differently from in `all-packages.nix`?

How should I move `transfig = fig2dev` to aliases.nix properly? @SuperSandro2000

**Comment by azuwis** [comment]
Result of `nixpkgs-review pr 156420` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>dblatexFull</li>
    <li>fig2dev</li>
    <li>kakounePlugins.rep</li>
    <li>xfig</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
> I'm confused, transfig = fig2dev in aliases.nix works differently from in all-packages.nix?
> How should I move transfig = fig2dev to aliases.nix properly? @SuperSandro2000

Yes, aliases.nix should not be used in nixpkgs but are still available for end users.

**Pull request merged**